src/plugins/inspector/common/adapters/request/request_adapter.ts

```diff
@@ -68,6 +87,50 @@ export class RequestAdapter extends EventEmitter {
     return Array.from(this.requests.values());
   }
 
+  /**
+   * Extract a string warning field from the json object
+   */
+  public extractWarnings(): ResponseWarning[] | undefined {
```

Comment:
Maybe I'm confused, but isn't this logic supposed to be the replacement logic for the error handling the `handleResponse` function is currently doing? It is looking for completely different things in the response (timeouts and shard failures): This can be refine later on, but I guess we want to extend this "warning gathering" logic - for example it seems like this is not capturing shard failures: https://github.com/elastic/kibana/pull/136833

https://github.com/elastic/kibana/blob/09a3fe42283583eaf640b1610466df194cf08b27/src/plugins/data/public/search/fetch/handle_response.tsx#L27

Reply:
Hi @flash1293 the current solution in the code is heavily based on how the [Search Example app was handling warnings](https://github.com/elastic/kibana/blob/8.4/examples/search_examples/public/search/app.tsx#L241-L246), since that was the easiest way for me to test. I need to back up and see about as you suggest, making the request adapter able to replace the [`handleResponse`](https://github.com/elastic/kibana/blob/09a3fe42283583eaf640b1610466df194cf08b27/src/plugins/data/public/search/fetch/handle_response.tsx#L27) logic.

Just curious, how are you simulating shard failures?

Reply:
This thread might be resolved per https://github.com/elastic/kibana/commit/27957111f35d2d37885bffa36351297fad0a2ee7